src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1330,6 +1331,32 @@ citus_update_node(PG_FUNCTION_ARGS)
 }
 
 
+Datum
+citus_pause_node(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	int32 nodeId = PG_GETARG_INT32(0);
+	List *placementList = NIL;
+
+	WorkerNode *workerNode = FindNodeAnyClusterByNodeId(nodeId);
+	if (workerNode == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_NO_DATA_FOUND),
+						errmsg("node %u not found", nodeId)));
+	}
+
+
+	if (NodeIsPrimary(workerNode))
```

Comment:
Let's extract the logic from citus_update_node and share it, this would also give us the possibility of adding `force`. It also allows us to later improve a single part of the code, to add support for pausing on worker nodes too.